src/Microsoft.AspNet.OData/Adapters/WebApiActionMap.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Web.Http.Controllers;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Interfaces;
+
+namespace Microsoft.AspNet.OData.Adapters
+{
+    /// <summary>
+    /// Adapter class to convert Asp.Net WebApi action map to OData WebApi.
+    /// </summary>
+    internal class WebApiActionMap : IWebApiActionMap
+    {
+        /// <summary>
+        /// The inner map wrapped by this instance.
+        /// </summary>
+        private ILookup<string, HttpActionDescriptor> innerMap;
+
+        /// <summary>
+        /// Initializes a new instance of the WebApiActionMap class.
+        /// </summary>
+        /// <param name="actionMap">The inner map.</param>
+        public WebApiActionMap(ILookup<string, HttpActionDescriptor> actionMap)
+        {
+            if (actionMap == null)
+            {
+                throw Error.ArgumentNull("actionMap");
+            }
+
+            this.innerMap = actionMap;
+        }
+
+        /// <summary>
+        /// Determines whether a specified key exists.
+        /// </summary>
+        /// <param name="name">The key.</param>
+        /// <returns>True if the key exist; false otherwise.</returns>
+        public bool Contains(string name)
+        {
+            return this.innerMap.Contains(name);
```

Comment:
How are the values (HttpActionDescriptor) of the map utilized? Should there be another method here for the HttpActionDescriptor?

Reply:
They get used at a higher layer which is AspNet-version specific. At the lower layers, i.e. where the adapter is used, the Contains check is used to validate that action name and then the action name is returned. The name is then used in the higher layer to get the action descriptor. ASPNet has churned in this area quite a bit, replacing a controller and action description with a single descriptor and changing the collection used to map them.